File: app/src/main/java/top/pressed/quickscaffold/basic/viewmodel/DataBindingViewModel.java
package top.pressed.quickscaffold.basic.viewmodel;

import android.content.Context;
import android.content.res.Resources;
import android.view.View;

import androidx.databinding.ViewDataBinding;
import androidx.lifecycle.ViewModel;

public abstract class DataBindingViewModel<VB extends ViewDataBinding> extends ViewModel {
    private VB binding;

    public VB getBinding() {
        return binding;
    }

    public void setBinding(VB binding) {
        this.binding = binding;
    }

    public View getRoot() {
        return binding.getRoot();
    }

    public Context getContext() {
        return getRoot().getContext();
    }

    public Resources.Theme getTheme() {
        return getContext().getTheme();
    }
}

File: app/src/main/java/top/pressed/quickscaffold/basic/viewmodel/DataBindingViewModel.kt
package top.pressed.quickscaffold.basic.viewmodel

import android.content.Context
import androidx.databinding.ViewDataBinding
import androidx.lifecycle.ViewModel
import android.content.res.Resources.Theme
import android.view.View

abstract class DataBindingViewModel<VB : ViewDataBinding> : ViewModel() {
    var binding: VB? = null
        private set

    fun setBinding(binding: VB) {
        this.binding = binding
    }

    val root: View
        get() = binding!!.root
    val context: Context
        get() = root.context
    val theme: Theme
        get() = context.theme
}